src/avatar_action.cpp

```diff
@@ -562,6 +562,14 @@ void avatar_action::swim( map &m, avatar &you, const tripoint &p )
     }
     you.setpos( p );
     g->update_map( you );
+
+    mtype_id mount_type;
+    if( you.is_mounted() ) {
+        mount_type = you.mounted_creature->type->id;
+    }
+    g->events().send<event_type::avatar_moves>( mount_type, m.ter( p ).id(),
+            you.get_movement_mode(), you.is_underwater() );
+
```

Comment:
This chunk of code is duplicated now; should probably be pulled out into a function.